app/javascript/packs/service_active_docs.js

```diff
@@ -1,8 +1,19 @@
 import { ActiveDocsSpecWrapper as ActiveDocsSpec } from 'ActiveDocs/components/ActiveDocsSpec'
+import { fetchData } from 'utilities/utils'
 
-document.addEventListener('DOMContentLoaded', () => {
+document.addEventListener('DOMContentLoaded', async () => {
   const containerId = 'swagger-ui-container'
-  const { url } = document.getElementById(containerId).dataset
+  const AUTOCOMPLETE_CONFIG = {
+    dataUrl: {
+      provider: '/p/admin/api_docs/account_data.json',
+      buyer: '/api_docs/account_data.json'
+    }
+  }
+  const { url, baseUrl, accountType } = document.getElementById(containerId).dataset
+  const dataUrl = accountType ? AUTOCOMPLETE_CONFIG.dataUrl[accountType] : 'buyer'
 
-  ActiveDocsSpec({ url }, containerId)
+  const data = await fetchData(`${baseUrl}${dataUrl}`)
```

Comment:
Could this block the UI for a long time?

Reply:
Well, trying in preview with cache disabled it takes about 1 second, I think it's acceptable 

<img width="1241" alt="Screenshot 2020-06-03 at 18 09 57" src="https://user-images.githubusercontent.com/13486237/83660739-777c5a00-a5c5-11ea-99df-05cb2aa745f6.png">

Reply:
But the thing is, it could slow it down more. Try using a slow 3G connection for instance (although I admit it's probably not the most common situation).
Using the promise instead would work around this [potential] problem.

Reply:
I will move the fetch inside `ActiveDocsSpec`, that way `swagger-ui` will show the loading spinner